### PR TITLE
DHFPROD-9333: Problem deleting concept class and relationship

### PR DESCRIPTION
**Pull request by eeduardo-marklogic**
DHFPROD-9329 Creating/Deleting relationship

#### Checklist: 
```diff
- Note: do not change the below
```

- ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
- [x] Ran cypress tests on firefox locally
  

- ##### Reviewer:

- [x] Reviewed Tests
- n/a Added to Release Wiki

### CODE REVIEW

**Review by MSBrignoni — APPROVED** [review]

**Review by briantang — DISMISSED** [review]

File: marklogic-data-hub-central/ui/e2e/cypress/e2e/modeling/verifyConceptClassesInModeling.cy.tsx [hunk 1]
```diff
@@ -365,4 +365,44 @@ describe("Concept classes in Modeling screen", () => {
     entityTypeTable.getConceptClass("TestConcept").should("not.exist");
   });
 
+  it("Create/Delete new relationship and new concept class from Table view", {defaultCommandTimeout: 120000}, () => {
```
Comment (aniamy) [inline marklogic-data-hub-central/ui/e2e/cypress/e2e/modeling/verifyConceptClassesInModeling.cy.tsx:368]: Any reason why we need to change the default timeout?

Comment (aniamy) [reply marklogic-data-hub-central/ui/e2e/cypress/e2e/modeling/verifyConceptClassesInModeling.cy.tsx:368]: Why not move this it before we delete the TestConcept? That past is already tested in the previous create and edit tests. And focus only on the relationship. Then the delete test cleans the concept.

File: marklogic-data-hub-central/ui/e2e/cypress/e2e/modeling/verifyConceptClassesInModeling.cy.tsx [hunk 2]
```diff
@@ -365,4 +365,44 @@ describe("Concept classes in Modeling screen", () => {
...
+      graphViewSidePanel.getConceptClassColor("TestConcept").should("have.css", "background-color", "rgb(213, 211, 221)");
+    }
+
+    modelPage.getIconSelected("TestConcept", "FaAccessibleIcon").should("exist");
+    conceptClassModal.getAddButton().click();
+    cy.waitForAsyncRequest();
+    conceptClassModal.getAddButton().should("not.exist");
+    cy.scrollTo("bottom");
+    table.scrollToFooter();
+    entityTypeTable.getConceptClass("TestConcept").should("exist").scrollIntoView();
```
Comment (aniamy) [inline marklogic-data-hub-central/ui/e2e/cypress/e2e/modeling/verifyConceptClassesInModeling.cy.tsx:396]: No need to tests all this part. The tests "Create/Edit and verify new concept class from Table view" is already doing all this checks.
Also I do not see the part were we create and edit a relationship related to a concept.

**Review by aniamy — CHANGES_REQUESTED** [review]
See comments

**Review by MSBrignoni — DISMISSED** [review]

File: marklogic-data-hub-central/ui/e2e/cypress/e2e/modeling/verifyConceptClassesInModeling.cy.tsx [hunk 3]
```diff
@@ -365,4 +364,130 @@ describe("Concept classes in Modeling screen", () => {
...
+    conceptClassModal.getAddButton().should("not.exist");
+
+    cy.wait(5000);
+    cy.log("**Create Concept Class**");
+    graphViewSidePanel.getSelectedConceptClassHeading("testDeleteStyle").should("exist");
+    graphViewSidePanel.getSelectedConceptClassHeadingInfo("testDeleteStyle").should("exist");
+    graphViewSidePanel.getConceptClassDescription().should("be.visible");
+    graphViewSidePanel.getConceptClassDescription().should("have.value", "Created just to validate deleting.");
+
+    modelPage.toggleColorSelector("testDeleteStyle");
```
Comment (aniamy) [inline marklogic-data-hub-central/ui/e2e/cypress/e2e/modeling/verifyConceptClassesInModeling.cy.tsx:384]: Again, I suggest we do this test one it above and re-use the Concept class alreadt created to optimize the times.
In this test we don't need to re-test things like picking colors and descriptions. We already covered this above.

File: marklogic-data-hub-central/ui/e2e/cypress/e2e/modeling/verifyConceptClassesInModeling.cy.tsx [hunk 4]
```diff
@@ -365,4 +364,130 @@ describe("Concept classes in Modeling screen", () => {
...
+      graphViewSidePanel.getConceptClassColor("testDeleteStyle").should("have.css", "background-color", "rgb(213, 211, 221)");
+    }
+
+    modelPage.openIconSelector("testDeleteStyle");
+    modelPage.selectIcon("testDeleteStyle", "FaAccessibleIcon");
+    modelPage.getIconSelected("testDeleteStyle", "FaAccessibleIcon").should("exist");
+    graphViewSidePanel.closeSidePanel();
+    graphViewSidePanel.getSelectedConceptClassHeading("testDeleteStyle").should("not.exist");
+
+    cy.log("**Create a relationship with the created concept class**");
```
Comment (aniamy) [inline marklogic-data-hub-central/ui/e2e/cypress/e2e/modeling/verifyConceptClassesInModeling.cy.tsx:399]: Suggestion, the it should start here using the concept we already have created in this file.

File: marklogic-data-hub-central/ui/e2e/cypress/e2e/modeling/verifyConceptClassesInModeling.cy.tsx [hunk 5]
```diff
@@ -365,4 +364,130 @@ describe("Concept classes in Modeling screen", () => {
...
+    cy.waitUntil(() => toolbar.getModelToolbarIcon()).click();
```
Comment (aniamy) [inline marklogic-data-hub-central/ui/e2e/cypress/e2e/modeling/verifyConceptClassesInModeling.cy.tsx:400]: We are already in the model tile. In this scenario there is no need to click the tile icon

File: marklogic-data-hub-central/ui/e2e/cypress/e2e/modeling/verifyConceptClassesInModeling.cy.tsx [hunk 6]
```diff
@@ -365,4 +364,130 @@ describe("Concept classes in Modeling screen", () => {
...
+    cy.waitForAsyncRequest();
+
+    graphVis.getPositionsOfNodes().then((nodePositions: any) => {
+      let testDeleteStyleCoordinates: any = nodePositions["testDeleteStyle"];
+      cy.wait(150);
+      cy.waitUntil(() => graphVis.getGraphVisCanvas().click(testDeleteStyleCoordinates.x, testDeleteStyleCoordinates.y, {force: true}));
+    });
+    cy.wait(1500);
+
+    graphViewSidePanel.getDeleteIcon("testDeleteStyle").scrollIntoView().click({force: true});
```
Comment (aniamy) [inline marklogic-data-hub-central/ui/e2e/cypress/e2e/modeling/verifyConceptClassesInModeling.cy.tsx:480]: Continuing with the suggestion: This part, deleting the concept, would be handled by the last it so we wouldn't need it.

**Review by aniamy — DISMISSED** [review]

**Review by ngodugu-marklogic — APPROVED** [review]

**Review by ngodugu-marklogic — DISMISSED** [review]

**Review by aniamy — APPROVED** [review]

**Review by MSBrignoni — APPROVED** [review]

**Pull request merged**